stdlib/public/Concurrency/TaskGroup.cpp

```diff
@@ -1641,8 +1652,9 @@ static void swift_taskGroup_waitAllImpl(
     ThrowingTaskFutureWaitContinuationFunction *resumeFunction,
     AsyncContext *rawContext) {
   auto waitingTask = swift_task_getCurrent();
-  waitingTask->ResumeTask = task_group_wait_resume_adapter;
-  waitingTask->ResumeContext = rawContext;
```

Comment:
this was incorrect, we cannot do this without holding the lock, as otherwise an `offer` may pull the waiting task off the queue; but only one of the paths shall update and run the task.